packages/runtime/container-runtime/src/pendingStateManager.ts

```diff
@@ -399,17 +398,14 @@ export class PendingStateManager implements IDisposable {
 
 			// Stringified content should match
 			if (pendingMessage.content !== messageContent) {
-				this.stateHandler.close(
-					DataProcessingError.create(
-						"pending local message content mismatch",
-						"unexpectedAckReceived",
-						message,
-						{
-							expectedMessageType: JSON.parse(pendingMessage.content).type,
-						},
-					),
+				throw DataProcessingError.create(
+					"pending local message content mismatch",
+					"unexpectedAckReceived",
+					message,
+					{
+						expectedMessageType: JSON.parse(pendingMessage.content).type,
```

Comment:
```suggestion
						messageType: message.type,
						expectedMessageType: JSON.parse(pendingMessage.content).type,
```
Would also be nice to have some extra telemetry